modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/WatchProcessor.java

```diff
@@ -284,16 +288,10 @@ private CompletableFuture<List<WatchAndEvents>> collectWatchesAndEvents(List<Ent
         }, watchExecutor);
     }
 
-    private CompletableFuture<Void> invokeOnRevisionCallback(List<WatchAndEvents> watchAndEventsList, long revision, HybridTimestamp time) {
+    private CompletableFuture<Void> invokeOnRevisionCallback(long revision, HybridTimestamp time) {
         try {
-            // Only notify about entries that have been accepted by at least one Watch.
-            var acceptedEntries = new HashSet<EntryEvent>();
-
-            for (WatchAndEvents watchAndEvents : watchAndEventsList) {
-                acceptedEntries.addAll(watchAndEvents.events);
-            }
-
-            var event = new WatchEvent(acceptedEntries, revision, time);
+            // We consciously put empty set here. Revision applied callback doesn't need any data.
```

Comment:
I would suggest to change the `revisionCallback` interface explicitly only require the revision and the timestamp

Reply:
Ok, I'll do it